### PR TITLE
Add additional verification for process execution

### PR DESCRIPTION
**Pull request by agocke**
Test failures seem to indicate that a file might be held by a still-running
process, but it doesn't appear that that is possible in the code. Verify
that the process is closed using an assert to double-check this assumption.

Hopefully helps with #13626

### CODE REVIEW

**Comment by agocke** [comment]
/cc @dotnet/roslyn-compiler @dotnet/roslyn-infrastructure 

And @jaredpar because I know how he feels about `File.Exists`

File: src/Test/Utilities/Desktop/ProcessUtilities.cs [hunk 1]
```diff
@@ -78,6 +78,8 @@ public static class ProcessUtilities
 
                 process.WaitForExit();
 
+                Debug.Assert(process.HasExited);
```
Comment (jasonmalinowski) [inline src/Test/Utilities/Desktop/ProcessUtilities.cs]: This means we're distrusting .WaitForExit? You might want a comment here. :smile:

File: src/Test/Utilities/Shared/TempFiles/DisposableFile.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public DisposableFile(string prefix = null, string extension = null, string dire
 
         public void Dispose()
         {
-            if (Path != null && File.Exists(Path))
+            if (Path != null)
```
Comment (jaredpar) [inline src/Test/Utilities/Shared/TempFiles/DisposableFile.cs]: Yay!

**Comment by agocke** [comment]
test windows_debug_unit32_prtest please

**Pull request merged**